minerl/Malmo/Minecraft/src/main/java/com/microsoft/Malmo/Mixins/MixinForgeHookStatsFix.java

```diff
@@ -0,0 +1,114 @@
+        world.captureBlockSnapshots = false;
+        if (ret == EnumActionResult.SUCCESS) {
+            int newMeta = itemstack.getItemDamage();
+            int newSize = itemstack.getCount();
+            NBTTagCompound newNBT = null;
+            if (itemstack.getTagCompound() != null) {
+                newNBT = itemstack.getTagCompound().copy();
+            }
+
+            BlockEvent.PlaceEvent placeEvent = null;
+            List<BlockSnapshot> blockSnapshots = (List)world.capturedBlockSnapshots.clone();
+            world.capturedBlockSnapshots.clear();
+            itemstack.setItemDamage(meta);
+            itemstack.setCount(size);
+            if (nbt != null) {
+                itemstack.setTagCompound(nbt);
+            }
+
+            if (blockSnapshots.size() > 1) {
+                placeEvent = ForgeEventFactory.onPlayerMultiBlockPlace(player, blockSnapshots, side, hand);
+            } else if (blockSnapshots.size() == 1) {
+                placeEvent = ForgeEventFactory.onPlayerBlockPlace(player, (BlockSnapshot)blockSnapshots.get(0), side, hand);
+            }
+
+            Iterator var18;
+            BlockSnapshot snap;
+            if (placeEvent != null && ((BlockEvent.PlaceEvent)placeEvent).isCanceled()) {
+                ret = EnumActionResult.FAIL;
+
+                for(var18 = blockSnapshots.iterator(); var18.hasNext(); world.restoringBlockSnapshots = false) {
+                    snap = (BlockSnapshot)var18.next();
+                    world.restoringBlockSnapshots = true;
+                    snap.restore(true, false);
+                }
+            } else {
+                itemstack.setItemDamage(newMeta);
+                itemstack.setCount(newSize);
+
+                if (nbt != null) {
+                    itemstack.setTagCompound(newNBT);
+                }
+
+                int updateFlag;
+                IBlockState oldBlock;
+                IBlockState newBlock;
+                for(var18 = blockSnapshots.iterator(); var18.hasNext(); world.markAndNotifyBlock(snap.getPos(), (Chunk)null, oldBlock, newBlock, updateFlag)) {
+                    snap = (BlockSnapshot)var18.next();
+                    updateFlag = snap.getFlag();
+                    oldBlock = snap.getReplacedBlock();
+                    newBlock = world.getBlockState(snap.getPos());
+                    if (!newBlock.getBlock().hasTileEntity(newBlock)) {
+                        newBlock.getBlock().onBlockAdded(world, snap.getPos(), newBlock);
+                    }
+                }
+
+                // BUGFIX: Get the actual item of the stack, regardless of whether the stack is empty
+                if (itemstack.getCount() == 0) itemstack.setCount(1);
+                player.addStat(StatList.getObjectUseStats(itemstack.getItem()));
+                itemstack.setCount(newSize);
+                // BUGFIX: end
+                // Original code:
+                // player.addStat(StatList.getObjectUseStats(itemstack.getItem()));
```

Comment:
If we were going to fully lean into mixins, this would be the perfect use case for the @inject annotation. Approving for now I will just leave a note to come back to this later

Reply:
I have re-implemented the fix using a @Redirect, which seemed more appropriate than @Inject after looking into it.